File: proiect_IO_python/engine/Searcher.py
import re

class SearcherClass:
    def __init__(self, filepath="", word=""):
        self.file = filepath
        self.word = word

    def setFile(self, filepath):
        self.file = filepath

    def setWord(self, word):
        self.word = word

    def getLine(self, index):
        try:
            line = ""

            print('Incerc sa citesc ', self.file, '...')
            reader = open(self.file, 'r')

            i = 0
            line = reader.readline()

            while line:
                if i==index:
                    reader.close()
                    return line

                i = i+1
                line = reader.readline()

            reader.close()
            if i<index:
                return "";

        except:
            print('N-o mers ;(')
            return ""

    def lastContaining(self):
        line = ""
        found = ""

        print('Incerc sa citesc ', self.file, '...')
        reader = open(self.file, 'r')

        line = reader.readline()
        while line:
            if re.search(".*\\s*"+self.word+"\\s*.*", line):
                found = line
            line = reader.readline()

        reader.close()

        return found

File: proiect_node/engine/searcher.js
const fs = require('fs');
const readline = require('readline');

exports.getLine = function(filepath, index, callback) {
    const fileStream = fs.createReadStream(filepath);

    const reader = readline.createInterface({
        input: fileStream
    });
    // Note: we use the crlfDelay option to recognize all instances of CR LF
    // ('\r\n') in input.txt as a single line break.

    var i = 0;
    let line = "";

    reader.on('line', (current) => {
        console.log("Linia "+i+": "+current);

        if(i==index)
            line = current;

        i++;

    });

    reader.on('close', () => {
        console.log("Linia dorita: "+line);

        callback(line);
    })
};

exports.getLast = function(filepath, word, callback) {
    const fileStream = fs.createReadStream(filepath);

    const reader = readline.createInterface({
        input: fileStream
    });
    // Note: we use the crlfDelay option to recognize all instances of CR LF
    // ('\r\n') in input.txt as a single line break.

    var i = 0;
    let line = "";

    reader.on('line', (current) => {
        console.log("Linia "+i+": "+current);

        if(current.match(".*\\s*"+word+"\\s*.*"))
            line = current;

        i++;
    });

    reader.on('close', () => {
        console.log("Linia dorita: "+line);

        callback(line);
    })
};